docs/docs/reconciliation.md

```diff
@@ -73,61 +67,56 @@ renderB: <div style={{'{{'}}fontWeight: 'bold'}} />
 
 After the attributes have been updated, we recurse on all the children.
 
-
 ### Custom Components
 
-We decided that the two custom components are the same. Since components are stateful, we cannot just use the new component and call it a day. React takes all the attributes from the new component and calls `componentWillReceiveProps()` and `componentWillUpdate()` on the previous one.
+The last case is comparing two custom components of the same type. Since components are stateful, we must keep the old instance around. React takes all the attributes from the new component and calls `componentWillReceiveProps()` and `componentWillUpdate()` on the previous one.
```

Comment:
This is not quite what I meant. 

Can we reorder cases and merge them?
I would like to see:

1. DOM Elements. First explain that if type differs they are replaced. Then explain how updates are applied if type is same.

2. Component Elements. First explain that if type differs old one is unmounted and new one is mounted. Then explain how componentWillReceiveProps is called if type is the same.

There is no need to mention custom components in section on DOM, and there is no need to have three sections. Two is enough and they should be symmetrical.